src/Command/UnregistrerAdherentsCommand.php

```diff
@@ -0,0 +1,175 @@
+            $output->writeln(
+                sprintf('%s file not found', $input->getArgument('fileUrl'))
+            );
+
+            return 1;
+        }
+
+        $this->em->beginTransaction();
+
+        if ($this->unregistrateAdherents($emails, $output)) {
+            return 1;
+        }
+
+        $this->em->commit();
+
+        $output->writeln('Unregistration finished.');
+
+        if (0 < count($this->emailNotFound)) {
+            $output->writeln('The following email adress are not found in DB :');
+            foreach ($this->emailNotFound as $email) {
+                $output->writeln($email);
+            }
+        }
+        if (0 < count($this->hosts)) {
+            $output->writeln('The adherents with following email adress are hosts of committees. They are not unregistrated :');
+            foreach ($this->hosts as $email) {
+                $output->writeln($email);
+            }
+        }
+        if (0 < count($this->citizenProjectAdministrators)) {
+            $output->writeln('The adherents with following email adress are administrators of citizen projects. They are not unregistrated :');
+            foreach ($this->citizenProjectAdministrators as $email) {
+                $output->writeln($email);
+            }
+        }
+    }
+
+    private function parseCSV(string $filename): array
+    {
+        $emails = [];
+        if (false === ($handle = fopen($filename, 'r'))) {
+            throw new FileNotFoundException(sprintf('% not found', $filename));
+        }
+
+        $isFirstRow = true;
+        while (false !== ($data = fgetcsv($handle, 100000, ';'))) {
+            if (true === $isFirstRow) {
+                $isFirstRow = false;
+
+                continue;
+            }
+
+            $row = array_map('trim', $data);
+            $emails[] = $row[4];
+        }
+        fclose($handle);
+
+        return $emails;
+    }
+
+    private function unregistrateAdherents(array $emails, OutputInterface $output): bool
+    {
+        $batchSize = 20;
+        $i = 1;
+        $progress = new ProgressBar($output, count($emails));
+        $progress->start();
+
+        foreach ($emails as $email) {
+            try {
+                if (null === ($adherent = $this->adherentRepository->findByEmail($email))) {
+                    $this->emailNotFound[] = $email;
+
+                    continue;
+                }
+
+                if ($adherent->isHost()) {
+                    $this->hosts[] = $email;
+
+                    continue;
+                }
+
+                if ($adherent->isAdministrator()) {
+                    $this->citizenProjectAdministrators[] = $email;
+
+                    continue;
+                }
+
+                $this->adherentManager->unregister($adherent, false);
+            } catch (\Exception $ex) {
+                $output->writeln(
+                    sprintf('An error occured while unregistering an adherent with email %s : %s', $email, $ex->getMessage())
```

Comment:
You could put some quotes around the email, no space before the colon.

Reply:
Done